plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/PodDeploymentPathProvider.java

```diff
@@ -0,0 +1,156 @@
+
+	private static final String DOCKER_IMAGE_DIGEST_IDENTIFIER = "sha256:";
+
+	public String load(IService service, Connection connection) throws CoreException {
+		IProject project = service.getProject();
+		IPod pod = getPod(service, project, connection);
+		String imageRef = getImageRef(connection, project, pod);
+		int imageDigestIndex = imageRef.indexOf(DOCKER_IMAGE_DIGEST_IDENTIFIER);
+		if (imageDigestIndex > 0) {
+			String imageDigest = imageRef.substring(imageDigestIndex);
+			IResource imageStreamTag = getImageStreamTag(connection, project, imageRef, imageDigest);
+			return getPodPath(imageStreamTag);
+		} else {
+			
+			return "";
+		}
+	}
+
+	private String getImageRef(Connection connection, IProject project, IPod pod) throws CoreException {
+		Collection<String> images = pod.getImages();
+		if (images.isEmpty()) {
+			throw new CoreException(OpenShiftCoreActivator.statusFactory()
+					.errorStatus(NLS.bind("No images found for pod {0} in project {1} on server {2}",
+							new Object[] { pod.getName(), project.getName(), connection.getHost() })));
+		}
+		// TODO: handle if there are 2+ images
+		String imageRef = images.iterator().next();
+		return imageRef;
+	}
+
+	private IPod getPod(IService service, IProject project, Connection connection) throws CoreException {
+		List<IPod> allPods = project.getResources(ResourceKind.POD);
+		List<IPod> pods = ResourceUtils.getPodsForService(service, allPods);
+		if (pods.isEmpty()) {
+			throw new CoreException(OpenShiftCoreActivator.statusFactory()
+					.errorStatus(NLS.bind("No pods found for service {0} in project {1} on server {2}. "
+							+ "Ensure your build is finished and a pod has been deployed.",
+							new Object[] { service.getName(), project.getName(), connection.getHost() })));
+		}
+		// TODO: handle if there are 2+ pods
+		IPod pod = pods.get(0);
+		return pod;
+	}
+
+	private IResource getImageStreamTag(Connection connection, IProject project, String imageRef, String imageDigest)
+			throws CoreException {
+		List<IResource> imageStreamTags = project.getResources(ResourceKind.IMAGE_STREAM_TAG);
+		IResource imageStreamTag = imageStreamTags.stream()
+				.filter(istag -> {
+					String imageName = ModelNode.fromJSONString(istag.toJson()).get("image").get("metadata").get("name").asString();
+					return imageDigest.equals(imageName);
+				})
+				.findFirst().orElse(null);
+		if (imageStreamTag == null) {
+			throw new CoreException(OpenShiftCoreActivator.statusFactory()
+					.errorStatus(NLS.bind("No metadata found for docker image {0} in project {1} on server {2}",
+							new Object[] { imageRef, project.getName(), connection.getHost() })));
+		}
+		// load image stream tag individually to get full ContainerConfig metadata
+		imageStreamTag = connection.getResource(ResourceKind.IMAGE_STREAM_TAG, project.getName(), imageStreamTag.getName());
+		return imageStreamTag;
+	}
+
+	private ImportImageMetaData getImageMetaData(String imageRef, IProject project) {
```

Comment:
good catch, that's intendetend to get used for docker images without sha256:XXX digests